### PR TITLE
Dashboard and assembly for configuration files

### PR DESCRIPTION
**Pull request by nohan-antony**

### CODE REVIEW

File: src/services/WolfDen.Infrastructure/Extensions/AddConventionsExtension.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+{
+    public static class AddConventionsExtension
+    {
+        public static void AddConventions(this ModelBuilder modelBuilder, string schemaName, Assembly? migrationAssembly)
+        {
+            modelBuilder.HasDefaultSchema(schemaName);
+            if (migrationAssembly != null)
+            {
+                modelBuilder.ApplyConfigurationsFromAssembly(migrationAssembly);
+            }
```
Comment (sanalsunnyGW) [inline src/services/WolfDen.Infrastructure/Extensions/AddConventionsExtension.cs:14]: Specify History Table

File: src/services/WolfDen.Infrastructure/Data/WolfDenContext.cs [hunk 2]
```diff
@@ -2,28 +2,57 @@
...
 using System.Text;
 using System.Threading.Tasks;
 using WolfDen.Domain.Entity;
 using WolfDen.Infrastructure.Configuration;
+using WolfDen.Infrastructure.Extensions;
 
 namespace WolfDen.Infrastructure.Data
 {
-    public class WolfDenContext:DbContext
+    public interface IDummyEntity
```
Comment (sanalsunnyGW) [inline src/services/WolfDen.Infrastructure/Data/WolfDenContext.cs:14]: Why ?

File: src/services/WolfDen.Infrastructure/Data/WolfDenContext.cs [hunk 3]
```diff
@@ -2,28 +2,57 @@
...
+            modelBuilder.AddConventions("dbo", Assembly.GetExecutingAssembly());
+
+
+            //History Table
+
+            foreach (var entityType in modelBuilder.Model.GetEntityTypes())
+            {
+                entityType.SetTableName(entityType.GetDefaultTableName());
+
+                if (!entityType.ClrType.IsAssignableTo(typeof(IDummyEntity)))
```
Comment (sanalsunnyGW) [inline src/services/WolfDen.Infrastructure/Data/WolfDenContext.cs:39]: Why this ?

File: src/services/WolfDen.Infrastructure/Data/WolfDenContext.cs [hunk 4]
```diff
@@ -2,28 +2,57 @@
...
+                {
+                    var historyTableName = $"{entityType.GetDefaultTableName()}_HT"; 
+                    entityType.SetHistoryTableName(historyTableName);
+                    entityType.SetHistoryTableSchema("dbo");
```
Comment (sanalsunnyGW) [inline src/services/WolfDen.Infrastructure/Data/WolfDenContext.cs:43]: only append schema with HT , you can keep the table name as such

**Review by sanalsunnyGW — APPROVED** [review]

**Review by Aravindh-sreenivas — APPROVED** [review]

**Pull request merged**